File: solutions/2026-04-26/solution.java
// Maximum Subarray
// Difficulty: Medium
// URL: https://leetcode.com/problems/maximum-subarray/

public int maxSubArray(int[] nums) {
    if (nums == null || nums.length == 0) {
        return 0;
    }
    int max_sum = Integer.MIN_VALUE;
    int cur_sum = 0;
    for (int num : nums) {
        cur_sum = Math.max(num, cur_sum + num);
        max_sum = Math.max(max_sum, cur_sum);
    }
    return max_sum;
}

File: solutions/2026-04-26/solution.py
# Maximum Subarray
# Difficulty: Medium
# URL: https://leetcode.com/problems/maximum-subarray/

def maxSubArray(nums):
    if not nums:
        return 0
    max_sum = float('-inf')
    cur_sum = 0
    for num in nums:
        cur_sum = max(num, cur_sum + num)
        max_sum = max(max_sum, cur_sum)
    return max_sum